server/src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -495,7 +495,9 @@ public void cleanFiles(
             final Store store = store();
             store.incRef();
             try {
-                store.cleanupAndVerify("recovery CleanFilesRequestHandler", sourceMetadata);
+                if (DiscoveryNode.isStateless(indexShard.indexSettings().getSettings()) == false || indexShard.routingEntry().primary()) {
```

Comment:
I think we need to pass in the node settings, not the index settings to use that method?

Reply:
Documentation of `IndexSettings#getSettings` mentions:
> Returns the settings for this index. These settings contain the node and index level settings where
 settings that are specified on both index and node level are overwritten by the index settings.

Reply:
So I think we're good here

Reply:
Sure, it will work, but seems more intuitive to use `indexShard.indexSettings().getNodeSettings()`?

Reply:
Oh sorry, I misunderstood your previous comment. Sure, `getNodeSettings` is more appropriate, thanks for pointing this.